proposal/string-split.md

```diff
@@ -0,0 +1,110 @@
+# Add split function to Strings Module: Draft 1
+
+*([Issue](https://github.com/sass/sass/issues/1950))*
+
+This proposal adds `string.split()` to the `sass:string` module.
+
+## Table of Contents
+
+* [Background](#background)
+* [Summary](#summary)
+* [Semantics](#semantics)
+  * [`string.split()`](#string.split)
+
+## Background
+
+> This section is non-normative.
+
+The `sass:string` module contains several functions for
+manipulating and finding out information about `strings`.
+Currently, though, there is no built-in function that splits
+one `string` into a list of substrings, and authors have been
+creating their own versions of functions that achieve this
+functionality.
+
+## Summary
+
+> This section is non-normative.
+
+This proposal adds the `string.split()` function to the
+`sass:string` module. The function takes a `string`, splits it
+based on a provided `separator`, and returns a space-separated
+list of substrings.
+
+This could be used to take a provided `string` and repurpose
+parts of it for some other use. For example, fonts 
+contained in a font stack list could be split off and 
+then used as keys in a new map. 
+
+Examples:
+
+  - ```scss
+    $fonts: "Helvetica Neue, Helvetica, Arial";
+    string.split($fonts, ', '); // "Helvetica Neue" "Helvetica" "Arial"
+    ```
+
+A third argument can limit the number of `strings` 
+returned in the list:
+
+- ```scss
+  string.split($fonts, ', ', 2); // "Helvetica Neue" "Helvetica"
```

Comment:
should the limit truncate the end of the string, or should it take all the remainder in the last part ? Javascript is one of the few languages implementing the truncation (which is easy to achieve otherwise by doing an unlimited split and slicing the array)

Reply:
@stof Are you saying, should this work by 1) making all the splits and then slicing the array at the limit or 2) only splitting up to the limit and then stopping? 

Do you think there would be downsides to taking one approach or the other?

Reply:
Well, to me, it would be more useful if `limit` would work by splitting the string on the `limit -1` first separators, with the last item containing all the remaining work.
Your current proposal does the opposite (truncating the remaining parts), and this could be achieved through an unlimited split and then slicing the list (but I just figured out that `list.slice` does not exist in Sass).

I remember reading an article a few weeks ago comparing the behavior of the string splitting API of lots of different languages in that regard (but unfortunately, I did not manage to find it back for now to share it here. But AFAIR, most languages are providing the non-truncating version of the API (Javascript being one of the exceptions).

Also, a benefit of the non-truncation is that we don't loose content when splitting the string, so joining the list items with the separator (not exposed as a function in `sass:list` either btw) always produces the original string.

Reply:
I'd lean towards matching the JS behavior here. @nex3: Do you have any thoughts on this?

Reply:
I agree with @stof here. I think JS's behavior is a misfeature, and we should match other languages in preserving the entire remainder of the string.

Reply:
In that case, let's change this to do that.

Reply:
@jathak I've updated this section (and the examples) to show `limit` working as noted in the conversation above. The function should preserve the remainder of the string now. 

@mirisuzanne made a function based on these steps that you can see [here](https://www.sassmeister.com/gist/19ec36ad3ee106ba98ad8af31a3577a4)